modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -553,6 +555,98 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         util::throw_error(std::logic_error("Only one video source is"
                                            " currently supported!"));
     }
+ 
+    // testing meta data is cv::GMatDesc{CV_8U,3,cv::Size{768,576}}
+    if (m_metas.empty()) {
+        // extract meta from first frame
+    } else {
+        //const auto& metas = m_metas;
+    }
+    
+    const auto& metas = m_metas;
+    
+    if (/*wasFinished*/true) {
+        auto pass_ctx = ade::passes::PassContext{*m_orig_graph.get()};
+        cv::gimpl::passes::initMeta(pass_ctx, metas);
+        //does inferMeta needed?? 
+        cv::gimpl::passes::inferMeta(pass_ctx, true);
+        //compile islands for m_orig_graph
+        cv::gimpl::passes::storeResultingMeta(pass_ctx);
+        // Get compileArgs from m_ops?? 
+        cv::gimpl::GCompiler::compileIslands(*m_orig_graph.get(), comp_args);
+        
+        GModel::Graph gm(*m_orig_graph);
+        
+        auto xtract_in = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec) {
+            const auto orig_data_nh
+                = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+            const auto &orig_data_info
+                = gm.metadata(orig_data_nh).get<Data>();
+            if (orig_data_info.shape == GShape::GARRAY) {
+                // FIXME: GArray lost host constructor problem
+                GAPI_Assert(!cv::util::holds_alternative<cv::util::monostate>(orig_data_info.ctor));
+            }
+            vec.emplace_back(RcDesc{ orig_data_info.rc
+                                   , orig_data_info.shape
+                                   , orig_data_info.ctor});
+        };
+        auto xtract_out = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec, cv::GMetaArgs &metas) {
+            const auto orig_data_nh
+                = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+            const auto &orig_data_info
+                = gm.metadata(orig_data_nh).get<Data>();
+            if (orig_data_info.shape == GShape::GARRAY) {
+                // FIXME: GArray lost host constructor problem
+                GAPI_Assert(!cv::util::holds_alternative<cv::util::monostate>(orig_data_info.ctor));
+            }
+            vec.emplace_back(RcDesc{ orig_data_info.rc
+                                   , orig_data_info.shape
+                                   , orig_data_info.ctor});
+            metas.emplace_back(orig_data_info.meta);
+        };
```

Comment:
Why do we need it here? Ins't it what is done in the constructor? What's the need to repeat this code here?

Reply:
Some code is deleted now.
Shod `xtract_in ` and `xtract_out` be implemented as a usual function??

Reply:
They are needed in a single place only, so it doesn't make a lot of sense